### PR TITLE
Add a first draft of Mesos documentation

### PR DESCRIPTION
**Pull request by chriskuehl**
This adds a high-level overview of how Mesos/Marathon/Zookeeper fit together in general (and at the OCF).

The plan is to later add sub-pages for each of these which discuss how to troubleshoot each component, as well as giving links to web UIs and general "how-to" topics.

Marathon would probably need more information about actually writing apps to be deployed by it, and how to deploy them.

### CODE REVIEW

**Comment by chriskuehl** [comment]
This is a start for #212

File: ocfweb/docs/docs/staff/backend/mesos.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+At the OCF, we have three Mesos masters:
+
+* `mesos0` (`whirlwind`, running on `jaws`)
+* `mesos1` (`pileup`, running on `pandemic`)
+* `mesos2` (`monsoon`, running on `hal`)
+
+At any given time, a Mesos cluster has only one *leader*, which is always one
+of the Mesos masters. At any given time, only one of the masters is the leader,
+and the other two masters are effectively doing no real work (but keeping track
```
Comment (Baisang) [inline ocfweb/docs/docs/staff/backend/mesos.md]: so the other two masters are just there to ensure higher availability? (i.e. one takes over when leader dies)

**Comment by Baisang** [comment]
i assume eventually we would also have docs just in general about containers and containerization? because that's something I think a lot of staff have less experience with as well.

though, this doc does a good job explaining things at that higher level without having to talk explicitly about containers and things like that, I think people will just be curious when they click on e.g. the Marathon link and see that it is "A container orchestration platform..."

**Comment by chriskuehl** [comment]
Probably we need some. But it's also hard to figure out where to draw the line at not reproducing documentation in our wiki that is well-covered elsewhere. This doc makes sense because it's about our infrastructure in particular, but we're not really using Docker any differently than other folks.

**Comment by chriskuehl** [comment]
(granted, some of the stuff in this document is definitely covered elsewhere, but I find the Mesos introduction really hard to parse -- it's difficult to figure out exactly what it's doing (what does "Program against your datacenter like it’s a single pool of resources" even mean?))
